packages/serverpod_test/lib/src/transaction_manager.dart

```diff
@@ -27,46 +34,55 @@ class TransactionManager {
     late Transaction localTransaction;
 
     unawaited(
-      serverpodSession.db.transaction((newTransaction) async {
-        localTransaction = newTransaction;
+      (serverpodSession.db as DatabaseProxy).transaction(
```

Comment:
Suggestion: Add private extention to serverpodSession that retrieve the proxy directly to remove the cast.

This is the only place we need to cast this right? Might as well just have a private extension.
I'm generally not a fan of casts so if we can avoid it, and it makes sense, I would rather pursue that option.